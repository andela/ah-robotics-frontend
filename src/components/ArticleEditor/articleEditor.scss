.right-column {
  text-align: right;
}

.cke_contents .cke_bottom {
  float: left;
  width: 100%;
}

.cke_reset {
  min-height: 400px;
  width: 100%;
}

.editor-input {
  width: 60%;
  margin-bottom: 1em;
  margin-top: 1em;
}

.editor-error {
  color: red;
}
.success-message{
color: #00d0a0;
}
.ui.green.button {
  margin: 1em;
  margin-right: -0.2em;
}