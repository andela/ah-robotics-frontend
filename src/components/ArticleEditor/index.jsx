import {
  Button, Container, Form, Grid,
} from 'semantic-ui-react';
import CKEditor from 'react-ckeditor-component';
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import './articleEditor.scss';

const ArticleEditor = ({
 onInputChange, onEditorChange, handleArticlePost, createArticle, body,
}) => (
  <Container>
    <Fragment>
      <Grid>
        <Grid.Row>
          <Grid.Column width={12}>
            <p
              className="editor-error"
            >
              {(createArticle.errors) && createArticle.errors.title && 'Title is required'}
            </p>
            <Form.Input
              fluid
              name="title"
              placeholder="Title"
              transparent
              className="editor-input"
              onChange={onInputChange}
              maxLength={80}
            />
            <p
              className="editor-error"
            >
              {(createArticle.errors) && createArticle.errors.description && 'Description is required'}
            </p>
            <Form.Input
              fluid
              name="description"
              placeholder="Description"
              transparent
              className="editor-input"
              onChange={onInputChange}
              required
              maxLength={80}
            />
            <p className="success-message">
              {(createArticle.isPosted) && 'Article posted successfully'}
            </p>
          </Grid.Column>
          <Grid.Column width={4} className="right-column">
            <Button
              type="submit"
              basic
              color="green"
              content="Post"
              position="right"
              onClick={handleArticlePost}
              loading={createArticle.isPosting}
            />
          </Grid.Column>
        </Grid.Row>
      </Grid>
      <div>
        {' '}
        <p className="editor-error">{(createArticle.errors) && createArticle.errors.body && 'Body cannot be empty'}</p>
      </div>
      <CKEditor content={body} events={{ change: onEditorChange }} />
    </Fragment>
  </Container>
);
ArticleEditor.propTypes = {
  onInputChange: PropTypes.func.isRequired,
  onEditorChange: PropTypes.func.isRequired,
  handleArticlePost: PropTypes.func.isRequired,
  createArticle: PropTypes.shape({}).isRequired,
  body: PropTypes.string.isRequired,
};
export default ArticleEditor;
