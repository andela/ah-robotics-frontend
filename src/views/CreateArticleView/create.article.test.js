import React from 'react';
import CreateArticle from './index';
import ArticleEditor from '../../components/ArticleEditor';

describe('test create article view', () => {
it('it render create article view', () => {
const wrapper = shallow(<CreateArticle />);
const handleArticlePost = jest.fn();
const onEditorChange = jest.fn();
const onInputChange = jest.fn();

expect(wrapper.length).toBe(1);
expect(wrapper.find(<ArticleEditor
  createArticle={{}}
  handleArticlePost={handleArticlePost}
  onEditorChange={onEditorChange}
  body=""
  onInputChange={onInputChange}
/>)).toBeDefined();
});
it('it should match snapshot', () => {
  const wrapper = shallow(<CreateArticle />);
  expect(wrapper).toMatchSnapshot();
});
});
