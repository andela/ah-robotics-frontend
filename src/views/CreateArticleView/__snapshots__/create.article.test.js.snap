// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test create article view it should match snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
